src/thread.cpp

```diff
@@ -1,227 +1,228 @@
 #include "common.hpp"
 #include "node.hpp"
 #include "random_test.hpp"
-#include <deque>
 #include <algorithm>
 #include <chrono>
 #include <cstring>
+#include <deque>
+#include <mutex>
 #include <random>
 #include <sstream>
-#include <mutex>
 
-// Atomic flag and variable for metadata loading
 std::atomic_flag lock_metadata = ATOMIC_FLAG_INIT;
 std::atomic<bool> metadata_loaded(false);
 
-// Mutex for thread-safe logging
 std::mutex general_log_mutex;
 
-// Get the number of affected rows safely
 inline unsigned long long Node::getAffectedRows(MYSQL *connection) {
-    if (mysql_affected_rows(connection) == ~(unsigned long long)0) {
-        return 0LL;
-    }
-    return mysql_affected_rows(connection);
+  if (mysql_affected_rows(connection) == ~(unsigned long long)0) {
+    return 0LL;
+  }
+  return mysql_affected_rows(connection);
 }
 
 void Node::workerThread(int number) {
-    std::ofstream thread_log;
-    std::ofstream client_log;
-
-    if (options->at(Option::LOG_CLIENT_OUTPUT)->getBool()) {
-        std::ostringstream cl;
-        cl << myParams.logdir << "/" << myParams.myName << "_step_"
-           << std::to_string(options->at(Option::STEP)->getInt()) << "_thread-"
-           << number << ".out";
-        client_log.open(cl.str(), std::ios::out | std::ios::trunc);
-        if (!client_log.is_open()) {
-            general_log << "Unable to open logfile for client output " << cl.str()
-                        << ": " << std::strerror(errno) << std::endl;
-            return;
-        }
-    }
-
-    bool log_all_queries = options->at(Option::LOG_ALL_QUERIES)->getBool();
-    size_t n_queries = 5;
-
-    if (options->at(Option::LOG_N_QUERIES) && options->at(Option::LOG_N_QUERIES)->getInt() > 0) {
-        n_queries = options->at(Option::LOG_N_QUERIES)->getInt();
+  std::ofstream thread_log;
+  std::ofstream client_log;
+
+  if (options->at(Option::LOG_CLIENT_OUTPUT)->getBool()) {
+    std::ostringstream cl;
+    cl << myParams.logdir << "/" << myParams.myName << "_step_"
+       << std::to_string(options->at(Option::STEP)->getInt()) << "_thread-"
+       << number << ".out";
+    client_log.open(cl.str(), std::ios::out | std::ios::trunc);
+    if (!client_log.is_open()) {
+      general_log << "Unable to open logfile for client output " << cl.str()
+                  << ": " << std::strerror(errno) << std::endl;
+      return;
     }
+  }
+
+  bool log_all_queries = options->at(Option::LOG_ALL_QUERIES)->getBool();
+  size_t n_queries = 5;
+
+  if (options->at(Option::LOG_N_QUERIES) &&
+      options->at(Option::LOG_N_QUERIES)->getInt() >= 0) {
```

Comment:
Logic Corrected for --log-N-queries=0.
Now it will save an empty log file in myParams.logdir i.e /tmp here.